hybridCarboniteMetadataService/src/main/scala/cromwell/services/metadata/hybridcarbonite/CarboniteWorkerActor.scala

```diff
@@ -1,47 +1,94 @@
 package cromwell.services.metadata.hybridcarbonite
 
 import akka.actor.{Actor, ActorLogging, ActorRef, Props}
+import cats.data.NonEmptyList
+import cromwell.core.retry.SimpleExponentialBackoff
+import cromwell.core.{WorkflowAborted, WorkflowFailed, WorkflowId, WorkflowSucceeded}
+import cromwell.services.metadata.MetadataArchiveStatus.Unarchived
+import cromwell.services.metadata.MetadataService.{QueryForWorkflowsMatchingParameters, WorkflowQueryFailure, WorkflowQuerySuccess}
+import cromwell.services.metadata.WorkflowQueryKey._
+import cromwell.services.metadata.hybridcarbonite.CarboniteWorkerActor.CarboniteWorkflowComplete
+import cromwell.services.metadata.hybridcarbonite.CarbonitingMetadataFreezerActor.FreezeMetadata
 import cromwell.util.GracefulShutdownHelper
 import cromwell.util.GracefulShutdownHelper.ShutdownCommand
-import cromwell.services.metadata.hybridcarbonite.CarboniteWorkerActor.DoCarboniting
 
 import scala.concurrent.ExecutionContext
 import scala.concurrent.duration._
 
-class CarboniteWorkerActor(carboniteInterval: FiniteDuration, ioActor: ActorRef) extends Actor with ActorLogging with GracefulShutdownHelper {
+class CarboniteWorkerActor(carboniterConfig: HybridCarboniteConfig,
+                           serviceRegistryActor: ActorRef,
+                           ioActor: ActorRef) extends Actor with ActorLogging with GracefulShutdownHelper {
 
   implicit val ec: ExecutionContext = context.dispatcher
 
+  val carboniteFreezerActor = context.actorOf(CarbonitingMetadataFreezerActor.props(carboniterConfig, self, serviceRegistryActor, ioActor))
+
+  val backOff = SimpleExponentialBackoff(
+    initialInterval = 5.seconds,
+    maxInterval = 5.minutes,
+    multiplier = 1.1,
+    randomizationFactor = 0.0
+  )
+
   scheduleNextCarboniting()
 
+
   override def receive: Receive = {
-    case DoCarboniting => doCarboniting()
-    case ShutdownCommand => stopGracefully()
-    case oops => log.error(s"Programmer Error! The CarboniteWorkerActor is not to be talked to! ($sender sent $oops})")
+    case s: WorkflowQuerySuccess => {
+      if (s.response.results.nonEmpty) carboniteWorkflow(s.response.results.head.id)
+      else scheduleNextCarboniting()
+    }
```

Comment:
TOL debatable suggestion, I'm curious what others think. I wanted to do something clever with a pattern match on the collection inside the response but it's a `Seq` and not a `List` 
```suggestion
    case s: WorkflowQuerySuccess if s.response.results.nonEmpty => carboniteWorkflow(s.response.results.head.id)
    case _: WorkflowQuerySuccess => scheduleNextCarboniting() 
```

Reply:
🤷‍♂ I'm sort of ambivalent... I'd just go for whichever you feel is more readable and tells a better code story. Or takes less "oh, *that*'s what's going on" time

Reply:
This is certainly clever. I have updated it